plugin/src/main/java/org/opensearch/ml/action/memorycontainer/memory/MemoryProcessingService.java

```diff
@@ -214,43 +189,18 @@ private List<String> parseFactsFromLLMResponse(MLOutput mlOutput) {
             return facts;
         }
 
-        for (int i = 0; i < modelTensors.getMlModelTensors().size(); i++) {
-            Map<String, ?> dataMap = modelTensors.getMlModelTensors().get(i).getDataAsMap();
-            if (dataMap != null && dataMap.containsKey("content")) {
-                try {
-                    List<?> contentList = (List<?>) dataMap.get("content");
-                    if (contentList != null && !contentList.isEmpty()) {
-                        Map<String, ?> contentItem = (Map<String, ?>) contentList.get(0);
-                        if (contentItem != null && contentItem.containsKey("text")) {
-                            String responseStr = String.valueOf(contentItem.get("text"));
-
-                            try (
-                                XContentParser parser = jsonXContent
-                                    .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, responseStr)
-                            ) {
-                                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-
-                                while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
-                                    String fieldName = parser.currentName();
-                                    if ("facts".equals(fieldName)) {
-                                        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser);
-                                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
-                                            String fact = parser.text();
-                                            facts.add(fact);
-                                        }
-                                    } else {
-                                        parser.skipChildren();
-                                    }
-                                }
-                            }
-                        }
-                    }
-                } catch (Exception e) {
-                    log.error("Failed to extract content from dataMap", e);
-                    throw new IllegalArgumentException("Failed to extract content from LLM response", e);
-                }
-                break;
+        // Parse the JSON response to extract facts
+        try {
+            Map<String, ?> dataMap = modelTensors.getMlModelTensors().get(0).getDataAsMap();
+            // if LLM Response does not contain FACTS_FIELD then fact extraction failed hence throw exception
+            if (dataMap == null || dataMap.get(FACTS_FIELD) == null) {
+                throw new IllegalArgumentException("Failed to parse facts from LLM response");
             }
+            facts = (List<String>) dataMap.get(FACTS_FIELD);
+        } catch (Exception e) {
```

Comment:
Here we are catching a generic exception and then throwing a 400 level exception. Can we try to catch specific exceptions to provide better error context?